kratos/solving_strategies/strategies/arc_length_strategy.h

```diff
@@ -0,0 +1,664 @@
+    typedef typename BaseType::TSystemVectorType TSystemVectorType;
+    typedef typename BaseType::LocalSystemVectorType LocalSystemVectorType;
+    typedef typename BaseType::LocalSystemMatrixType LocalSystemMatrixType;
+    typedef typename BaseType::TSystemMatrixPointerType TSystemMatrixPointerType;
+    typedef typename BaseType::TSystemVectorPointerType TSystemVectorPointerType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    // Constructor
+    ArcLengthStrategy(
+        ModelPart& model_part,
+        typename TSchemeType::Pointer pScheme,
+        typename TConvergenceCriteriaType::Pointer pNewConvergenceCriteria,
+        typename TBuilderAndSolverType::Pointer pNewBuilderAndSolver,
+        Parameters ThisParameters
+        ) : ResidualBasedNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>(model_part, pScheme,
+                pNewConvergenceCriteria, pNewBuilderAndSolver, ThisParameters["max_iteration"].GetInt(),
+                ThisParameters["compute_reactions"].GetBool(), ThisParameters["reform_dofs_at_each_step"].GetBool(),
+                ThisParameters["move_mesh_flag"].GetBool())
+        {
+            ThisParameters = this->ValidateAndAssignParameters(ThisParameters, this->GetDefaultParameters());
+            AssignSettings(ThisParameters);
+        }
+
+    /**
+     * @brief Destructor.
+     * @details In trilinos third party library, the linear solver's preconditioner should be freed before the system matrix. We control the deallocation order with Clear().
+     */
+    ~ArcLengthStrategy() override = default;
+
+
+    /**
+     * @brief This method assigns settings to member variables
+     * @param ThisParameters Parameters that are assigned to the member variables
+     */
+    void AssignSettings(const Parameters ThisParameters) override
+    {
+        ModelPart& r_model_part = BaseType::GetModelPart();
+        BaseType::AssignSettings(ThisParameters);
+        mDesiredIterations = ThisParameters["desired_iterations"].GetInt();
+        mMaxRadiusFactor   = ThisParameters["max_radius_factor"].GetDouble();
+        mMinRadiusFactor   = ThisParameters["min_radius_factor"].GetDouble();
+        mInitializeArcLengthWasPerformed = false;
+
+        // we initialize the list of load processes to be taken into account
+        if (ThisParameters["loads_sub_model_part_list"].size() > 0) {
+            mSubModelPartList.resize(ThisParameters["loads_sub_model_part_list"].size());
+            mVariableNames.resize(ThisParameters["loads_variable_list"].size());
+
+            if (mSubModelPartList.size() != mVariableNames.size())
+                KRATOS_THROW_ERROR( std::logic_error, "For each SubModelPart there must be a corresponding nodal Variable", "")
+
+            for (unsigned int i = 0; i < mVariableNames.size(); i++) {
+                mSubModelPartList[i] = &( r_model_part.GetSubModelPart(ThisParameters["loads_sub_model_part_list"][i].GetString()));
```

Comment:
This is quite slow, a Parameters is a slow acess class.

Use GetStringArray() instead